backend/api/endpoints/hint.py

```diff
@@ -0,0 +1,23 @@
+from fastapi import APIRouter, Response, Cookie
+from typing import Optional
+
+from api.logic import user_management
+from routes import Hint
+from db_funcs import hintFuncs
+
+router = APIRouter()
+
+@router.post("/api/hint/{hint_id}")
+def get_hint(hint_id: int, response: Response, session_cookie: Optional[str] = Cookie(None)):
+    user = user_management.get_user_by_token(session_cookie)
+    if user is None:
+        response.status_code = 401
+        return {"result": "user is not authorized"}
+
+    hint = hintFuncs.get_hint(hint_id)
+    if hint is None:
+        response.status_code = 204
+        return {"result": "missing hint"}
+    # use time cost || score cost
```

Comment:
Ještě potřebujeme zanést to, že team si šifru zobrazil, abychom potom mohli počítat penalizace. Na to jsme zapomněli, a chybí nám v DB tabulka. Dořeším

Reply:
Jakmile bude mergnutý #67 tak to je, bude to v tabulce hint_used (hint_id, team_id)

Reply:
Taak, a je to připraveno. Je potřeba, při otevření, zanést do tabulky dvojici hint_id, team_id daného člověka. Pokud už neexistuje! Pokud ano, měla by vyvstat chyba. Tu je možné ignorovat.